File: leetcode/CountNicePairsInAnArray_1814.go
package leetcode

func reverseNum(num int) (revNum int) {
	for num != 0 {
		revNum = (revNum * 10) + (num % 10)
		num /= 10
	}
	return
}

func countNicePairs(nums []int) int {
	var m map[int]int = make(map[int]int)
	var count int = 0

	for x := range nums {
		x = x + reverseNum(x)
		count += m[x]
		m[x] += 1
	}

	return count
}

File: leetcode/CountNicePairsInAnArray_1814.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
    int reverse(int num) {
        int rnum = 0;
        while (num != 0) {
            rnum *= 10 + (num % 10);
            num /= 10;
        }
        return rnum;
    }

public:
    int countNicePairs(vector<int>& nums) {
        unordered_map<int, int> m;
        int count = 0;

        for(int ele: nums) {
            ele = ele - reverse(ele);
            count += m[ele];
            m[ele]++;
        }

        return count;
    }
};